src/array/binary/mutable.rs

```diff
@@ -105,12 +105,13 @@ impl<O: Offset> MutableBinaryArray<O> {
         }
     }
 
-    /// Reserves `additional` slots.
-    pub fn reserve(&mut self, additional: usize) {
+    /// Reserves `additional` elements and `additional_values` on the values buffer.
+    pub fn reserve(&mut self, additional: usize, additional_values: usize) {
```

Comment:
do we still need this change? imo we can keep it, but the I would suggest a separate PR to handle since it is backward incompatible.

Reply:
I think it's better to keep the same api like `MutableUtf8`.

Reply:
Let's do it in another pr.